x-pack/plugin/core/src/main/resources/ecs-template.yml

```diff
@@ -0,0 +1,222 @@
+      # No path matching is needed as it is assumed these fields are already in the json doc a long
+      # Elasticsearch would do this mapping automatically but it is specified here for documentation purpose
+      # DISCUSS: If a field "7" as string comes in, will it still be mapped correctly?
+      - long_as_long:
+          match_mapping_type: long
+          runtime:
+            type: long
+
+      # Matching all long and double fields
+      # This matches the following fields: event.risk_score
+      # No path matching is needed as it is assumed these fields are already in the json doc a float or double
+      # Elasticsearch would do this mapping automatically but it is specified here for documentation purpose
+      # DISCUSS: If a field "7.7" as string comes in, will it still be mapped correctly?
+      - double_as_double:
+          match_mapping_type: double
+          runtime:
+            type: double
+
+      # Matching all boolean fields
+      # This matches the following fields: exists
+      # No path matching is needed as it is assumed these fields are already in the json doc as boolean
+      # Elasticsearch would do this mapping automatically but it is specified here for documentation purpose
+      # DISCUSS: If a field "true" as string comes in, will it still be mapped correctly?
+      - strings_as_boolean:
+          match_mapping_type: boolean
+          runtime:
+            type: boolean
+
+      # Matching all string fields by default to keyword
+      # TODO: Does first / last match win in the list?
+      - strings_as_keyword:
+          match_mapping_type: string
+          runtime:
+            type: keyword
+            # Do we need this for runtime?
+            ignore_above: 1024
+
+      # Any message field except the top level one is matched as text but not indexed
+      # TODO: Does this work? Which one has priority?
+      - match_message:
+          path_match: message
+          runtime:
+            type: match_only_text
+
+      # Match all ip fields to ip
+      - match_ip:
+          match_mapping_type: string
+          # DISCUSS: If match could be an array, these mappings could be written much more compact
+          match: ip
+          runtime:
+            type: ip
+
+      # Match all forwarded_ip fields to ip. This is for network.forwarded_ip
+      - match_ip:
+          match_mapping_type: string
+          match: forwarded_ip
+          runtime:
+            type: ip
+
+      # Match all labels fields. Flattened type instead of keyword is used to prevent too many fields to be created.
+      # DISCUSS: Is this the right choice?
+      - match_labels:
+          match_mapping_type: string
+          # TODO: Does this also match the top level labels?
+          match: labels
+          runtime:
+            type: flattened
+
+      # Match location fields to geopoint
+      - match_location:
+          # TODO: What is the correct matching type here?
+          match_mapping_type: object
```

Comment:
nit: when you have name conventions, I think you can safely drop the match_mapping_type

Reply:
Agree, I had it more as a safeguard. What will happen, if a field is `location` and it is keyword instead of an object? As it is a runtime field, ES will just ignore it as conversion fails or is this the script error case?